modules/solid_mechanics/src/materials/SingleVariableReturnMappingSolution.C

```diff
@@ -181,116 +182,156 @@ SingleVariableReturnMappingSolutionTempl<is_ad>::internalSolve(
+                     scalar_lower_bound,
+                     iter_output);
+
+      if (converged(_residual, reference_residual))
       {
-        if (residual_old - _residual != 0.0)
-        {
-          GenericReal<is_ad> alpha = residual_old / (residual_old - _residual);
-          alpha = MathUtils::clamp(alpha, 1.0e-2, 1.0);
+        outputIterationStep(iter_output, effective_trial_stress, scalar, reference_residual);
+        break;
+      }
+      else
+      {
+        bool modified_increment = false;
 
-          if (alpha != 1.0)
+        // Line Search
+        if (_line_search)
+        {
+          if (residual_old - _residual != 0.0)
           {
-            modified_increment = true;
-            scalar_increment *= alpha;
-            if (iter_output)
-              *iter_output << "  Line search alpha = " << MetaPhysicL::raw_value(alpha)
-                           << " increment = " << MetaPhysicL::raw_value(scalar_increment)
-                           << std::endl;
+            GenericReal<is_ad> alpha = residual_old / (residual_old - _residual);
+            alpha = MathUtils::clamp(alpha, 1.0e-2, 1.0);
+
+            if (alpha != 1.0)
+            {
+              modified_increment = true;
+              scalar_increment *= alpha;
+              if (iter_output)
+                *iter_output << "  Line search alpha = " << MetaPhysicL::raw_value(alpha)
+                             << " increment = " << MetaPhysicL::raw_value(scalar_increment)
+                             << std::endl;
+            }
           }
         }
-      }
 
-      if (_bracket_solution)
-      {
-        // Check to see whether trial scalar_increment is outside the bounds, and set it to a point
-        // within the bounds if it is
-        if (scalar_old + scalar_increment >= scalar_upper_bound ||
-            scalar_old + scalar_increment <= scalar_lower_bound)
+        if (_bracket_solution)
         {
-          if (scalar_upper_bound != max_permissible_scalar &&
-              scalar_lower_bound != min_permissible_scalar)
+          // Check to see whether trial scalar_increment is outside the bounds, and set it to a
+          // point within the bounds if it is
+          if (scalar_old + scalar_increment >= scalar_upper_bound ||
+              scalar_old + scalar_increment <= scalar_lower_bound)
           {
-            const Real frac = 0.5;
-            scalar_increment =
-                (1.0 - frac) * scalar_lower_bound + frac * scalar_upper_bound - scalar_old;
-            modified_increment = true;
-            if (iter_output)
-              *iter_output << "  Trial scalar_increment exceeded bounds.  Setting between "
-                              "lower/upper bounds. frac: "
-                           << frac << std::endl;
+            if (scalar_upper_bound != max_permissible_scalar &&
+                scalar_lower_bound != min_permissible_scalar)
+            {
+              const Real frac = 0.5;
```

Comment:
Is there any benefit to this change here, or is this about transitioning to a consistent use of `constexpr` because tehre are cases where it might make a difference?

Reply:
the latter. It's not a big deal as you point out